api/python/provisioner/commands/__init__.py

```diff
@@ -685,6 +698,82 @@ def run(self, targets):  # noqa: C901 FIXME
             ) from update_exc
 
 
+@attr.s(auto_attribs=True)
+class SWRollback(CommandParserFillerMixin):
+    input_type: Type[inputs.NoParams] = inputs.NoParams
+    _run_args_type = RunArgsSWRollback
+
+    @staticmethod
+    def _rollback_component(component, targets):
+        state_name = "components.{}.rollback".format(component)
+        try:
+            logger.info(
+                "Rolling back {} on {}".format(component, targets)
+            )
+            StatesApplier.apply([state_name], targets)
+        except Exception:
+            logger.exception(
+                "Failed to rollback {} on {}".format(component, targets)
+            )
+            raise
+
+    def run(self, target_version, targets):
+
+        local_minion = local_minion_id()
+        rollback_path = PillarKey('rollback')
+        rollback_pillar = PillarResolver(local_minion).get([rollback_path])
+        rollback_pillar = rollback_pillar[local_minion][rollback_path]
+        cortx_tgt_version = (
+            target_version
+            if target_version in rollback_pillar.keys()
+            else None
+        )
+
+        if cortx_tgt_version is None:
+            raise ValueError(
+                f'{target_version} version data not available for rollback'
+            )
+        else:
+            txn_id_path = PillarKey(f"rollback/{cortx_tgt_version}")
+            txn_id_pillar = PillarResolver(local_minion).get([txn_id_path])
+            txn_id_pillar = next(iter(txn_id_pillar.values()))
+            try:
+                for target in txn_id_pillar[txn_id_path]:
+                    txn_id = txn_id_pillar[txn_id_path][target]['yum_txn_id']
+
+                    if not txn_id or txn_id is values.MISSED:
+                        raise BadPillarDataError(
+                            f"yum txn id not available for {target}"
+                        )
+                    else:
+                        logger.info(f"Starting rollback on target {target}")
+                        salt_cmd_run(
+                            f"yum history rollback -y {txn_id}",
+                            targets=target
+                        )
+                        logger.info(
+                            f"Rollback on target {target} is completed"
```

Comment:
Please adjust the message to `Yum rollback...` since at that moment rollback is not fully completed yet, only in scope of yum packages. Later when everything is done completely for a node makes sense to add some INFO level message (like that one). And one more for the full set of nodes at the final end.

Reply:
Addressed